File: problem_solving_algo_and_ds/python/data_structures/divide_by_2.py
from stack import *
def divide_by_2(num):
    stack = Stack()

    while num > 0:
        rem = num % 2
        stack.push(rem)
        num = num // 2

    cov_num = ""
    while not stack.isEmpty():
        cov_num += str(stack.pop())

    return cov_num

print(divide_by_2(12))

File: problem_solving_algo_and_ds/ruby/data_structures/divide_by_2.rb
require_relative 'stack'

def divide_by_2(num)
  stack = Stack.new

  while num > 0
    rem = num % 2
    stack.push(rem)
    num = num / 2
  end

  conv_num = ""

  while !stack.is_empty
    conv_num += stack.pop.to_s
  end

  return conv_num
end

p divide_by_2(12)
